example/src/middlewares/auth.middleware.ts

```diff
@@ -18,5 +18,8 @@ const unauthorized$ = (req$: Observable<HttpRequest>) =>
 
 export const authorize$: Effect<HttpRequest> = request$ =>
   request$.pipe(
-    switchMap(() => merge(authorized$(request$), unauthorized$(request$))),
+    switchMap(() => merge(
+      authorized$(request$),
```

Comment:
and about the [iff](https://github.com/ReactiveX/rxjs/blob/master/src/internal/observable/iif.ts) method?

```javascript
request$.pipe(
    switchMap((req) => iif(
        () => isAuthorized(req),
        of(req),
        throwError(new HttpError('Unauthorized', HttpStatus.UNAUTHORIZED))
    )),
  );
```

Reply:
Thats cool. I supposed that this method was removed from RxJS API. IMO there is a big mess and they finally need to clean this up.